src/class/Cars.js

```diff
@@ -1,60 +1,98 @@
 import Car from "./Car";
+import Validator from "./Validator";
 
-const CAR_ADVANCE_MAX_NUMBER = 9;
+export default class Cars {
+  #cars = [];
+  #DEFAULT_RACING_ROUND_NUMBER = 5;
+  #roundNumber = this.#DEFAULT_RACING_ROUND_NUMBER;
+  #advanceConditions;
+  #validateRoundNumber = Validator.validateRoundNumber;
+  #validateCarNames = Validator.validateCarNames;
+  GAME_STEP = Object.freeze({
+    SET_CARS: "SET_CARS",
+    SET_ROUND_NUMBER: "SET_ROUND_NUMBER",
+    EXECUTE_ROUND: "EXECUTE_ROUND",
+    AWARDS: "AWARDS",
+  });
+  #CAR_NAME_SEPARATOR = ",";
+  #nextGameStep = this.GAME_STEP.SET_CARS;
 
-const CAR_ADVANCE_THRESHOLD_NUMBER = 4;
+  constructor(advanceConditions) {
+    this.#advanceConditions =
+      advanceConditions instanceof Array &&
+      advanceConditions.every(
+        (advanceCondition) => typeof advanceCondition === "function",
+      )
+        ? advanceConditions
+        : [];
+  }
 
-const DEFAULT_RACING_ROUND_NUMBER = 5;
+  addCar(name) {
+    this.#validateCarNames([name], this.getCarNames());
 
-const defaultAdvanceCondition = () => {
-  return Math.random() * CAR_ADVANCE_MAX_NUMBER >= CAR_ADVANCE_THRESHOLD_NUMBER;
-};
+    const car = new Car(name);
 
-export default class Cars {
-  cars;
-  advanceCondition;
-  roundNumber;
+    this.#cars.push(car);
+  }
+
+  addCars(names) {
+    this.#validateCarNames(names, this.getCarNames());
 
-  constructor(advanceCondition = defaultAdvanceCondition) {
-    this.cars = [];
-    this.advanceCondition = advanceCondition;
-    this.roundNumber = DEFAULT_RACING_ROUND_NUMBER;
+    names.forEach((car) => this.addCar(car));
+
+    this.#nextGameStep = this.GAME_STEP.SET_ROUND_NUMBER;
   }
 
-  addCar(name) {
-    const car = new Car(name, this.advanceCondition);
+  initializeCarsFromString(input) {
+    const carNameArray = input.split(this.#CAR_NAME_SEPARATOR);
 
-    this.cars.push(car);
+    this.addCars(carNameArray);
   }
 
   getAllCarStatus() {
-    return this.cars.map((car) => ({
-      name: car.getName(),
-      distance: car.getDistance(),
+    return this.#cars.map((car) => ({
+      name: car.name,
+      distance: car.distance,
     }));
   }
 
-  executeOneRound() {
-    this.cars.forEach((car) => car.advance());
+  getCarNames() {
+    return this.#cars.map((car) => car.name);
+  }
```

Comment:
```suggestion
  get carNames() {
    return this.#cars.map((car) => car.name);
  }
```
이렇게 getter를 사용해보면 좋을 것 같아요!